bookshelf/app/models/user.rb

```diff
@@ -0,0 +1,129 @@
+require 'bcrypt'
+# Talentica - bootstrap project
+# Copyright   Ashwini Kumar
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+class User < ActiveRecord::Base
+  include BCrypt
+  extend MyModules::OaouthImage
+  has_many :authentications,
+           class_name: 'UserAuthentication', dependent: :destroy
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable
+  devise :database_authenticatable, :registerable, :confirmable,
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable, omniauth_providers:  [:google_oauth2, :facebook]
+  has_one :profile, dependent: :destroy
+  has_many :posts,  dependent: :destroy
+  has_many :roles
+  after_create :set_profile
+  has_attached_file :avatar,
+                    styles: { medium: '300x300>', thumb: '50x50>' },
+                    path: 'public/system/:class/:id/:style/:filename',
+                    url: '/system/:class/:id/:style/:basename.:extension',
+                    default_url: '/images/no-image.png'
+
+  validates_attachment_content_type :avatar, content_type: %r{\Aimage\/.*\Z}
+
+  # before_save :extract_dimensions
+  scope :listingUser, -> (param) { where('firstname LIKE ?', "#{param}%") }
+
+  scope :email_validator, -> (param) { where('email = ?', param) }
+
+  def self.create_from_omniauth(params)
+    fname, lname = split_names params
+    @user = User.new(email:  params['info']['email'],
+                     firstname: fname,
+                     lastname: lname,
+                     avatar_file_name: 'img.jpg',
+                     password: params['info']['email'],
+                     confirmed_at: Time.now)
+    # don't send email to the user while signup with externa devise
+    @user.skip_confirmation!
+    @user.save
+    user_oaouth_image params.info.image, @user if params.info.image
+    @user.confirm!
+    @user
+  end
+
+  def self.split_names(param)
+    if param['provider'] == 'facebook'
+      param['info']['name'].split(' ', 2)
+    elsif param['provider'] == 'google_oauth2'
+      [param['info']['first_name'], param['info']['last_name']]
+    end
+  end
+
+  # devise confirm! method overriden
+  def confirm!
+    welcome_message
+    super
+  end
+
+  # creating a user by admin
+  def self.from_admin(params)
```

Comment:
`from_admin` doesn't seem to be the correct name, because:
1] The method knows nothing about 'admin'
2] This is just model, it shouldn't worry much about access and from where its method get called.

A better name would be just `create_user`, it's our job to ensure from controller that only admin user gets to call this method.